src/librustc_resolve/diagnostics.rs

```diff
@@ -948,6 +948,33 @@ use something_which_doesnt_exist;
 Please verify you didn't misspell the import's name.
 "##,
 
+E0434: r##"
+A dynamic environment was used inside a fn item. Example of erroneous code: 
+
+
+```compile_fail
+fn foo(){
+    let y = 5;
+    fn bar() -> u32 {
+        y // error: can't capture dynamic environment in a fn item; use the 
+          // || { ... } closure form instead.
```

Comment:
Please align the second line to the beginning of the error message at the first line:

```Rust
       y // error: can't capture dynamic environment in a fn item; use the 
         //        || { ... } closure form instead.
```